### DISCUSSION TITLE
AttributeError: 'function' object has no attribute 'orderbook'

### DISCUSSION THREAD

**Discussion by Jeraz** [ssantoshp/Empyrial#105] [Q&A] [answered]
I am unable to get the orderbook. Can someone help me with the error? Thanks
Is orderbook function supported?

Error: AttributeError: 'function' object has no attribute 'orderbook'

From empyrial import empyrial, Engine , get_report

portfolio = Engine(    
                  start_date= "2018-01-01", #start date for the backtesting
                  #end_date= "2023-05-07", #end date for the backtesting
                  portfolio= ["VGT", "SCHD", "VOO", "VOOV","VHT","VTI","TQQQ","QQQ"], #assets in your portfolio 
                  #weights = [0.2, 0.2, 0.2, 0.2, 0.2], #equal weighting is set by default
                  benchmark = ["SPY"], #SPY is set by default
                  optimizer = "HRP", # defines Mean-Variance as the optimizer
                 # max_vol = 0.25, #maximize the return for this level of volatility (25%)
                  #min_weights = 0.01, #invest at least 5% of the capital in every assets
     		  #max_weights = 0.95, #don't invest more than 35% in one asset
     		  #rebalance = ["2018-04-30", "2019-04-30", "2020-04-30", "2021-04-30","2022-04-30","2023-04-30"]
     		  rebalance = "1y" #rebalance every year
)
empyrial.orderbook
get_report(portfolio)

**Comment by santoshlite** [comment, marked as answer]
Hey, that's because of two things: 
1. `get_report` doesn't generate an orderbook but `empyrial(portfolio)` does.
2. You should call `empyrial.orderbook` after calling the function empyrial.

Here is the fix: https://colab.research.google.com/drive/1p-vclwEXJo06orBkkno7Qvj3hJP8a8Kc?usp=sharing

**Reply by Jeraz** [reply]
Thanks, that helps!
